tests/functional/targets/test_performance.py

```diff
@@ -0,0 +1,130 @@
+
+
+@given("admin provided the token to the API requester", target_fixture="token")
+def token(access_token):
+    return access_token
+
+
+@given(
+    "the API requester has generated the 'headers' with the token",
+    target_fixture="headers",
+)
+def generate_headers(token):
+    header_token = {"Authorization": f"Bearer {token}"}
+    return header_token
+
+
+@given(
+    parse(
+        "the API requester sends {num_requests} requests with {num_targets} "
+        "targets to RSTUF"
+    ),
+    target_fixture="multiple_requests",
+)
+def send_requests_with_targets(
+    headers, num_requests, num_targets, http_request
+):
+    def multiple_targets(number_of_targets):
+        targets = []
+        for count in range(0, int(number_of_targets)):
+            filename = f"test/{names_generator.generate_name()}-{count}.tar.gz"
+            target = {
+                "info": {
+                    "length": random.randint(999, 99999),
+                    "hashes": {
+                        "blake2b-256": blake2b(
+                            filename.encode(), digest_size=32
+                        ).hexdigest()
+                    },
+                },
+                "path": filename,
+            }
+            targets.append(target)
+
+        result = http_request(
+            "POST",
+            headers=headers,
+            url="/api/v1/targets",
+            json={"targets": targets},
+        )
+        return result
+
+    count = 0
+    tasks = []
+    while count < int(num_requests):
+        task_result = multiple_targets(number_of_targets=int(num_targets))
+        task_data = task_result.json()["data"]
+        tasks.append(
+            {
+                "task_id": task_data["task_id"],
+                "num_targets": len(task_data["targets"]),
+                "targets": task_data["targets"],
+                "last_update": task_data.get("last_update", None),
+            }
+        )
+        count += 1
+
+    return tasks
+
+
+@when(
+    parse(
+        "the API requester expects task 'SUCCESS' and status 'Task finished.' "
+        "before {timeout} seconds"
+    ),
+    target_fixture="tasks_result",
+)
+def get_tasks(headers, multiple_requests, http_request, timeout):
+    tasks = multiple_requests
+    tasks_result = []
+    while len(tasks) > 0:
+        time.sleep(3)
+        for task in tasks:
+            total_time_now = datetime.utcnow() - dateutil.parser.parse(
+                task["last_update"]
+            )
+            assert total_time_now.total_seconds() <= int(timeout)
+            response = http_request(
+                "GET",
+                headers=headers,
+                url=f"/api/v1/task/?task_id={task['task_id']}",
+            )
+
+            data = response.json()["data"]
+            state = data.get("state", None)
+            if state == "SUCCESS":
+                result = data.get("result", {})
+                if result.get("status", "") == "Task finished.":
```

Comment:
Btw: what do you win by providing a default value of `""` when using `result.get()` here?
If you don't provide any default value it will be `None` which is fine for this case.

Reply:
In a discussion with Kairo, he mentioned that he preferred to check a string to `Task finished.` in order to make one simple check instead of checking if the return value of `get` is `None` and if not it's `Task finished.`